base/docker-integration.go

```diff
@@ -0,0 +1,313 @@
+package base
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/couchbase/gocb"
+	"github.com/ory/dockertest"
+	"github.com/ory/dockertest/docker"
+)
+
+const (
+	dockerRepository = "couchbase"
+	dockerTag        = "enterprise-6.0.1"
+)
+
+//Structs used when checking health of nodes and buckets
+type nodes struct {
+	Status string `json:"status"`
+	Uptime string `json:"uptime"`
+}
+
+type responseBody struct {
+	Nodes []nodes `json:"nodes"`
+}
+
+//This will start up a Docker Instance of Couchbase Server and will be used to run tests against
```

Comment:
Could do with documenting the usage of this a bit better.

```suggestion
// NewDockerTest will start up a Docker container running Couchbase Server and will be used by tests to run against. Each Go package must declare its own TestMain which calls this function.
// Usage:
// func TestMain(m *testing.M) {
// 	base.NewDockerTest(m)
// } 
```